src/prefect/migration_helper.py

```diff
@@ -0,0 +1,58 @@
+import sys
+
+
+class ModuleMovedError(ImportError):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+MOVED_MODULES = {
+    "prefect.filesystems.GCS": "prefect_gcp",
+    "prefect.filesystems.Azure": "prefect_azure",
+}
+
+
+def handle_moved_modules(module_name, moved_modules):
+    """
+    Handle imports for moved or removed modules.
+
+    This function creates a custom __getattr__ function to intercept attribute access
+    on the given module and raise appropriate errors if the module has been moved or removed.
+
+    Args:
+        module_name (str): The name of the module to handle.
+        moved_modules (dict): A dictionary mapping old module paths to new locations or removal messages.
+
+    Usage:
+    Add this snippet to the top of the module that contains moved or removed modules.
+
+    Be sure to update the `MOVED_MODULES` dictionary with any old module paths and their new locations.
+    ```python
+    from prefect.migration_helper import handle_moved_modules, MOVED_MODULES
+
+    handle_moved_modules(__name__, MOVED_MODULES)
+    ```
+    """
+
+    def __getattr__(name):
+        qualified_name = f"{module_name}.{name}"
+        # Skip special attributes like __path__
+        if name.startswith("__") and name.endswith("__"):
+            raise AttributeError
+
+        # Check if the attribute name corresponds to a moved or removed module
+        if qualified_name in moved_modules:
+            new_location = moved_modules[qualified_name]
+            if "removed" in new_location:
+                raise ModuleMovedError(
+                    f"Module '{qualified_name}' has been removed. {new_location.split('use ')[-1]}"
```

Comment:
Sorry how does the removal case work? If i put in {"prefect.task_runners.SequentialTaskRunner": "removed"} what is the resultant error exactly?

On the margin, in the removal case, I think we may consider making the message a little less automated, i may want to link docs, qualify the removal, etc.

Reply:
lol you addressed my comment with the commit that came a minute after this comment. thanks.

Reply:
It would be for you to decide what to put there. Hopefully this comment is helpful enough otherwise I can update it: 
https://github.com/PrefectHQ/prefect/pull/13816/files#diff-8851b31434b3e0b3633cb9c295155a73dfd65a437f630a7f71fa90ff8f0c55b1R17-R20

Example:
```bash
In [2]: from prefect.filesystems import Azure
---------------------------------------------------------------------------
ModuleMovedError                          Traceback (most recent call last)
Cell In[2], line 1
----> 1 from prefect.filesystems import Azure

File ~/code-oss/prefect/src/prefect/utilities/migration_helper.py:74, in handle_moved_modules.<locals>.CustomModule.__getattr__(self, name)
     73 def __getattr__(self, name):
---> 74     return __getattr__(name)

File ~/code-oss/prefect/src/prefect/utilities/migration_helper.py:59, in handle_moved_modules.<locals>.__getattr__(name)
     57 new_location = moved_modules[qualified_name]
     58 if "removed" in new_location.lower():
---> 59     raise ModuleMovedError(
     60         f"Module '{qualified_name}' has been removed. {new_location.split('Removed: ')[-1]}"
     61     )
     62 else:
     63     raise ModuleMovedError(
     64         f"Module '{qualified_name}' has been moved to '{new_location}'. Please update your import."
     65     )

ModuleMovedError: Module 'prefect.filesystems.Azure' has been removed. Use 'prefect_azure' instead
```

Reply:
Hm, this should actually say `prefect.filesystems` has been removed though, so I need to address that

Reply:
In this case, this is inaccurate because it's not a module, so I may need to infer the type  to ensure the output message is right (and the error type)